crates/context/config/src/client.rs

```diff
@@ -201,13 +203,55 @@ impl Client<AnyTransport> {
             }
         }
 
+        let mut evm_transport = None;
+
+        'skipped: {
+            if let Some(evm_config) = config.signer.local.protocols.get("evm") {
+                let Some(e) = config.params.get("evm") else {
+                    eyre::bail!("missing config specification for `{}` signer", "evm");
+                };
+
+                if !matches!(e.signer, ClientSelectedSigner::Local) {
+                    break 'skipped;
+                }
+
+                let mut config = evm::EvmConfig {
+                    networks: Default::default(),
+                };
+                println!("evm_config: {:?}", evm_config);
+                for (network, signer) in &evm_config.signers {
+                    let Credentials::Raw(credentials) = &signer.credentials else {
+                        eyre::bail!("expected EVM credentials but got {:?}", signer.credentials)
+                    };
+
+                    let Some(account_id) = &credentials.account_id else {
+                        eyre::bail!("missing account id for `{}` signer", network);
+                    };
+
+                    let _ignored = config.networks.insert(
+                        network.clone().into(),
+                        evm::NetworkConfig {
+                            rpc_url: signer.rpc_url.clone(),
+                            account_id: account_id.clone(),
+                            access_key: credentials.secret_key.clone(),
+                        },
+                    );
+                }
+
+                evm_transport = Some(evm::EvmTransport::new(&config));
+            }
+        }
+
         let all_transports = Both {
             left: near_transport,
             right: Both {
                 left: starknet_transport,
                 right: Both {
                     left: icp_transport,
-                    right: stellar_transport,
+                    right: Both {
+                        left: stellar_transport,
+                        right: evm_transport,
+                    },
                 },
             },
         };
```

Comment:
would be interesting to introduce a macro to simplify this at some point

```rust
transport! {
    pub type LocalTransports<'a> = (
        MaybeNear<'a>,
        MaybeStarknet<'a>,
        MaybeIcp<'a>,
        MaybeStellar<'a>,
    );
};
```

and

```rust
let all_transport = transport!(
    near_transport,
    starknet_transport,
    icp_transport,
    stellar_transport,
);
```

Reply:
Filed in #1150.